apps/protocol-envio/src/utils/framework.ts

```diff
@@ -10,33 +10,134 @@ import {
   polygon,
   scroll,
   sepolia,
+  Chain,
 } from "viem/chains";
 import { Address } from "../types";
 import ERC20 from "../../abis/ERC20.json";
 import PRBProxy from "../../abis/PRBProxy.json";
 import PRBProxyRegistry from "../../abis/PRBProxyRegistry.json";
 import ERC20Bytes from "../../abis/ERC20Bytes.json";
 
-const supported = [
-  arbitrum,
-  avalanche,
-  base,
-  bsc,
-  gnosis,
-  mainnet,
-  optimism,
-  polygon,
-  scroll,
-  sepolia,
+type networks = { chain: Chain; rpcEndpoints: Array<string> };
+
+/** all rpc endpoints are public */
+const supported: Array<networks> = [
+  {
+    chain: arbitrum,
+    rpcEndpoints: [
+      "https://arbitrum.llamarpc.com",
+      "https://arb-pokt.nodies.app",
+      "https://endpoints.omniatech.io/v1/arbitrum/one/public",
+      "https://rpc.tornadoeth.cash/arbitrum",
+      "https://rpc.ankr.com/arbitrum",
```

Comment:
I'm not sure the best place to put this in your code base, I started in constants/chain but feel thats more used for the config templating